indexed_gzip/zran.c

```diff
@@ -644,16 +645,26 @@ int zran_init(zran_index_t *index,
     /*
      * Calculate the size of the compressed file
      */
-    if (fseek_(fd, f, 0, SEEK_END) != 0)
-        goto fail;
+    if (seekable_(fd, f)) {
+        if (fseek_(fd, f, 0, SEEK_END) != 0)
+            goto fail;
 
-    compressed_size = ftell_(fd, f);
+        compressed_size = ftell_(fd, f);
 
-    if (compressed_size < 0)
-        goto fail;
+        if (compressed_size < 0)
+            goto fail;
 
-    if (fseek_(fd, f, 0, SEEK_SET) != 0)
-        goto fail;
+        if (fseek_(fd, f, 0, SEEK_SET) != 0)
+            goto fail;
+    } else {
+        /*
+         * File is not seekable, so don't calculate
+         * compressed_size.  It will be updated in
+         * _zran_read_data_from_file when the EOF
+         * is reached
+         */
+        compressed_size = 0;
```

Comment:
`compressed size == 0` means that it is not yet known; it will be set when EOF is reached, at which point it will be used for bounds checking, and validation when importing a previously exported index (as the compressed size is saved to the index file).